src/triage/experiments/multicore.py

```diff
@@ -103,13 +110,19 @@ def parallelize(partially_bound_function, tasks, n_processes):
     num_successes = 0
     num_failures = 0
     results = []
-    with Pool(n_processes, maxtasksperchild=1) as pool:
-        for result in pool.map(partially_bound_function, tasks):
-            if result:
+    with ProcessPool(n_processes, max_tasks=1) as pool:
+        future = pool.map(partially_bound_function, tasks)
+        iterator = future.result()
+        results = []
+        while True:
+            try:
+                results.append(next(iterator))
                 num_successes += 1
-            else:
+            except StopIteration:
+                break
+            except Exception:
                 num_failures += 1
-            results.append(result)
+                logging.exception('Child failure')
```

Comment:
I really prefer the clarity of try/except/else but NBD:

    try:
        result = next(iterator)
    except StopIteration:
        break
    except Exception:
        logging.exception(...)
        num_failures += 1
    else:
        results.append(result)
        num_successes += 1